nav2_bringup/launch/nav2_bringup_launch.py

```diff
@@ -14,118 +14,85 @@
 
 import os
 
-from ament_index_python.packages import get_package_prefix
-from ament_index_python.packages import get_package_share_directory
-from launch.conditions import IfCondition
-
-import launch.actions
-import launch_ros.actions
+from ament_index_python.packages import get_package_prefix, get_package_share_directory
 
-from nav2_common.launch import RewrittenYaml
+from launch import LaunchDescription
+from launch.launch_description_sources import PythonLaunchDescriptionSource
+from launch.conditions import IfCondition
+from launch.substitutions import LaunchConfiguration, ThisLaunchFileDir
+from launch.actions import DeclareLaunchArgument, SetEnvironmentVariable, IncludeLaunchDescription
+from launch_ros.actions import Node
 
 
 def generate_launch_description():
     # Get the launch directory
-    launch_dir = os.path.join(get_package_share_directory('nav2_bringup'), 'launch')
+    bringup_dir = get_package_share_directory('nav2_bringup')
+    launch_dir = os.path.join(bringup_dir, 'launch')
 
     # Create the launch configuration variables
-    map_yaml_file = launch.substitutions.LaunchConfiguration('map')
-    use_sim_time = launch.substitutions.LaunchConfiguration('use_sim_time')
-    params_file = launch.substitutions.LaunchConfiguration('params')
-    bt_xml_file = launch.substitutions.LaunchConfiguration('bt_xml_file')
-    autostart = launch.substitutions.LaunchConfiguration('autostart')
+    map_yaml_file = LaunchConfiguration('map_yaml_file')
+    use_sim_time = LaunchConfiguration('use_sim_time')
+    params_file = LaunchConfiguration('params_file')
+    bt_xml_file = LaunchConfiguration('bt_xml_file')
+    autostart = LaunchConfiguration('autostart')
 
-    stdout_linebuf_envvar = launch.actions.SetEnvironmentVariable(
+    stdout_linebuf_envvar = SetEnvironmentVariable(
         'RCUTILS_CONSOLE_STDOUT_LINE_BUFFERED', '1')
 
-    # Create our own temporary YAML files that include substitutions
-    param_substitutions = {
-        'use_sim_time': use_sim_time,
-        'yaml_filename': map_yaml_file,
-        'bt_xml_filename': bt_xml_file,
-        'autostart': autostart
-    }
-
-    configured_params = RewrittenYaml(
-        source_file=params_file, rewrites=param_substitutions,
-        convert_types=True)
-
     # Declare the launch arguments
-    declare_map_yaml_cmd = launch.actions.DeclareLaunchArgument(
-        'map',
-        default_value=os.path.join(launch_dir, 'turtlebot3_world.yaml'),
+    declare_map_yaml_cmd = DeclareLaunchArgument(
+        'map_yaml_file',
+        default_value=os.path.join(bringup_dir, 'map', 'turtlebot3_world.yaml'),
```

Comment:
Can we get rid of this default? I think it should always be given as a parameter.

Reply:
OK